lib/client/receiveddata.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var _ = require('lodash');
+
+function mergeDataUpdate(isDelta, cachedDataArray, receivedDataArray) {
+
+  function nsArrayDiff(oldArray, newArray) {
+    var seen = {};
+    var l = oldArray.length;
+    for (var i = 0; i < l; i++) { seen[oldArray[i].mills] = true }
+    var result = [];
+    l = newArray.length;
+    for (var j = 0; j < l; j++) {
+      if (!seen.hasOwnProperty(newArray[j].mills)) {
+        result.push(newArray[j]); console.log('delta data found');
+      }
+    }
+    return result;
+  }
+
+  // If there was no delta data, just return the original data
+  if (!receivedDataArray) {
+    return cachedDataArray || [];
+  }
+
+  // If this is not a delta update, replace all data
+  if (!isDelta) {
+    return receivedDataArray || [];
+  }
+
+  // If this is delta, calculate the difference, merge and sort
+  var diff = nsArrayDiff(cachedDataArray, receivedDataArray);
+  return cachedDataArray.concat(diff).sort(function(a, b) {
+    return a.mills - b.mills;
+  });
+}
+
+function mergeTreatmentUpdate(isDelta, cachedDataArray, receivedDataArray) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1705074224/issues/source?bid=2452374&fileBranchId=3046616#l38)